src/Rector/AbstractRector.php

```diff
@@ -332,6 +376,27 @@ protected function removeNode(Node $node): void
         $this->nodeRemover->removeNode($node);
     }
 
+    private function isStmtPHPStanUnreachable(Stmt $stmt): bool
+    {
+        if ($stmt->getAttribute(AttributeKey::IS_UNREACHABLE) === true) {
+            // here the scope is never available for next stmt so we have nothing to refresh
+            return true;
+        }
+
+        $parentStmt = $stmt;
+        while ($parentStmt = $parentStmt->getAttribute(AttributeKey::PREVIOUS_NODE)) {
```

Comment:
It lookup previous stmt , should variable named previousStmt ?

Reply:
do you think it won't drop performance ?

Reply:
Without this check, rules will fail as unreachable stmts have no scope.

Any ideas how to speed the check up?

Reply:
PR https://github.com/rectorphp/rector-src/pull/2295 should already handle multiple rules applied, and scope note exist, so if that AbstractScope aware usecase; this check should be no longer necessary